Code/Framework/AzToolsFramework/AzToolsFramework/AssetDatabase/AssetDatabaseConnection.cpp

```diff
@@ -2929,6 +2947,52 @@ namespace AzToolsFramework
                 return GetResult(callName, statement, handler);
             }
 
+            bool GetWildcardSourceDependencyResult(const char* callName, SQLite::Statement* statement, AssetDatabaseConnection::sourceFileDependencyHandler handler)
+            {
+                AZ_UNUSED(callName); // AZ_Error may be compiled out entirely in release builds.
+
+                Statement::SqlStatus result = statement->Step();
+
+                SourceFileDependencyEntry entry;
+                int absolutePathMatch;
+                auto boundColumns = CombineColumns(entry.GetColumns(), MakeColumns(MakeColumn("AbsolutePath", absolutePathMatch)));
+
+                bool validResult = result == Statement::SqlDone;
+                while (result == Statement::SqlOK)
+                {
+                    if (!boundColumns.Fetch(statement))
+                    {
+                        return false;
+                    }
+
+                    // Its possible for certain relative wildcard dependencies to match an absolute path (ex c%.foo matches c:/bar.foo)
```

Comment:
I'm a little confused how we can tell whether c%.foo is relative or absolute wild card dependencies in this case. Is there a scenario where we do want to match absolute path with this pattern?

Reply:
c%.foo is assumed to be relative because its very unlikely someone would make a wildcard where they say "I want any file on the c drive with the .foo extension AND I'm not going to say c:/"  If it had c:/ it would be considered an absolute wildcard (when passed to the path class and we ask "is this absolute" it returns true).

The other part which may be confusing about this is that the query itself has a special return column that indicates whether we matched with the absolute or relative path.  Basically for the query we say "select all dependencies that match <absolute>, <relative>, or <uuid>".  I set up the query to return a 1 when the dependsOnSource == <absolute>, 0 otherwise.  That way we know the wildcard match was against the absolute path we input.  Keep in mind here that "dependsOnSource" is the wildcard query in this case (its backwards from how you'd usually do things).

Hopefully that made sense?